guests/eth-block/src/main.rs

```diff
@@ -36,8 +76,50 @@ pub fn main() {
         .expect("Failed to execute transactions")
         .build::<BuildFromMemDbStrategy>()
         .expect("Failed to build the resulting block");
-    // Output the resulting block's hash to the journal
-    env::commit(&output.hash());
-    // Leak memory, save cycles
-    core::mem::forget(output);
+
+    // get hash of the block header and print it
+
+    let output_hash = output.hash();
+    let output_hash_str = output_hash.to_string();
+    println!("{}", output_hash_str);
+
+    // generate random Ethereum public & private keys
+
+    use secp256k1::rand::rngs::OsRng;
+    use secp256k1::{Secp256k1, Message};
+    use secp256k1::hashes::sha256;
+
+    let secp = Secp256k1::new();
+    let (secret_key, public_key) = secp.generate_keypair(&mut OsRng);
+
+    // sign the above hash of the block header with the above secret key
+
+    let message = Message::from_hashed_data::<sha256::Hash>(output_hash_str.as_bytes());
+    let sig = secp.sign_ecdsa(&message, &secret_key);
+    // eprintln!("Private key: {}", secret_key.);
+    eprintln!("Public key: 0x{}", public_key);
+    eprintln!("Signature: 0x{}", sig);
+    assert!(secp.verify_ecdsa(&message, &sig, &public_key).is_ok());
+
+    Ok(())
 }
+
+// Sample output of this binary:
+//
+// ubuntu@VM-0-6-ubuntu:~/zeth-john/guests/eth-block/target/debug$ gramine-sgx ./eth-block --file /tmp/162.json.gz 
+// Gramine is starting. Parsing TOML manifest file, this may take some time...
+// -----------------------------------------------------------------------------------------------------------------------
+// Gramine detected the following insecure configurations:
+//
+//   - sgx.debug = true                           (this is a debug enclave)
+//   - loader.insecure__use_cmdline_argv = true   (forwarding command-line args from untrusted host to the app)
+//   - sys.insecure__allow_eventfd = true         (host-based eventfd is enabled)
+//   - sgx.allowed_files = [ ... ]                (some files are passed through from untrusted host without verification)
```

Comment:
So using files as way to communicate is generally not secure? Or it just has to be done in a different way?

Reply:
I think it's perfectly fine/safe to use files as input in our case. Alternatively, we can read input files from standard input, but using files as input should also work fine. See this thread where I discussed this concern with Gramine's lead developer: https://github.com/gramineproject/gramine/discussions/1580